go/vt/dbconfigs/dbconfigs.go

```diff
@@ -24,191 +26,256 @@ import (
-		}
+	data, err := json.MarshalIndent(out, "", "  ")
+	if err != nil {
+		return err.Error()
 	}
-	if DbaConfig&flags != 0 {
-		if err := initConnParams(&dbConfigs.Dba, socketFile); err != nil {
-			return nil, fmt.Errorf("dba dbconfig cannot be initialized: %v", err)
-		}
+	return string(data)
+}
+
+// Copy returns a copy of the DBConfig.
+func (dbcfgs *DBConfigs) Copy() *DBConfigs {
+	result := &DBConfigs{userConfigs: make(map[string]*userConfig)}
+	for k, u := range dbcfgs.userConfigs {
+		newu := *u
+		result.userConfigs[k] = &newu
 	}
-	if FilteredConfig&flags != 0 {
-		if err := initConnParams(&dbConfigs.Filtered, socketFile); err != nil {
-			return nil, fmt.Errorf("filtered dbconfig cannot be initialized: %v", err)
+	result.DBName.Set(dbcfgs.DBName.Get())
+	result.SidecarDBName.Set(dbcfgs.SidecarDBName.Get())
+	return result
+}
+
+// Init will initialize all the necessary connection parameters.
+// Precedence is as follows: if baseConfig command line options are
+// set, they supersede all other settings.
+// If baseConfig is not set, the next priority is with per-user connection
+// parameters. This is only for legacy support.
+// If no per-user parameters are supplied, then the defaultSocketFile
+// is used to initialize the per-user conn params.
+func Init(defaultSocketFile string) (*DBConfigs, error) {
+	// This is to support legacy behavior: use supplied socket value
+	// if conn parameters are not specified.
+	// TODO(sougou): deprecate.
+	for _, uc := range dbConfigs.userConfigs {
+		if uc.param.UnixSocket == "" && uc.param.Host == "" {
+			uc.param.UnixSocket = defaultSocketFile
 		}
 	}
-	if ReplConfig&flags != 0 {
-		if err := initConnParams(&dbConfigs.Repl, socketFile); err != nil {
-			return nil, fmt.Errorf("repl dbconfig cannot be initialized: %v", err)
+
+	// The new base configs, if set, supersede legacy settings.
+	if baseConfig.Host != "" || baseConfig.UnixSocket != "" {
+		for _, uc := range dbConfigs.userConfigs {
+			uc.param.Host = baseConfig.Host
+			uc.param.Port = baseConfig.Port
+			uc.param.UnixSocket = baseConfig.UnixSocket
+			uc.param.Charset = baseConfig.Charset
+			uc.param.Flags = baseConfig.Flags
+			if uc.useSSL {
+				uc.param.SslCa = baseConfig.SslCa
+				uc.param.SslCaPath = baseConfig.SslCaPath
+				uc.param.SslCert = baseConfig.SslCert
+				uc.param.SslKey = baseConfig.SslKey
+			}
 		}
 	}
-	// the Dba connection is not linked to a specific database
-	// (allows us to create them)
-	if dbConfigs.Dba.DbName != "" {
-		log.Warningf("dba dbname is set to '%v', ignoring the value", dbConfigs.Dba.DbName)
-		dbConfigs.Dba.DbName = ""
+
+	// See if the CredentialsServer is working. We do not use the
+	// result for anything, this is just a check.
+	for _, uc := range dbConfigs.userConfigs {
+		if _, err := WithCredentials(&uc.param); err != nil {
+			return nil, fmt.Errorf("dbconfig cannot be initialized: %v", err)
+		}
+		// Check for only one.
+		break
 	}
+	dbConfigs.SidecarDBName.Set("_vt")
```

Comment:
Use the command line parameter rather than hard-coding "_vt".

Reply:
It's the eventual intent.

The problem is that "_vt" is hard-coded all over the place. Before we can do the command-line, we need to refactor those places to use the sidecar name field.